ddraw/IDirect3DDeviceX.cpp

```diff
@@ -330,6 +330,28 @@ HRESULT m_IDirect3DDeviceX::SetTransform(D3DTRANSFORMSTATETYPE dtstTransformStat
 			break;
 		}
 
+		if(Config.DdrawConvertHomogeneousW)
+		{
+			if(dtstTransformStateType == D3DTS_VIEW)
```

Comment:
assumes that SetTransform is called, which is not the case for DK2 for example

Reply:
> assumes that SetTransform is called, which is not the case for DK2 for example

Agreed.  Maybe it is better to call `GetTransform()` and then `SetTransform()` after that inside the Draw function?

Also, another concern is if a game calls `GetTransform()`, then shouldn't we put the D3DMatrix back before sending to the application?  Maybe we should store the `SetTransform()` and then send it back when the application asks for it.